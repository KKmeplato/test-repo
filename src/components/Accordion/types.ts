import {
  AccordionItemProps as RadixAccordionItemProps,
  AccordionMultipleProps as RadixAccordionMultipleProps,
  AccordionSingleProps as RadixAccordionSingleProps
} from "@radix-ui/react-accordion";

export interface AccordionItemProps extends RadixAccordionItemProps {
  actionText?: string;
  children?: string | React.ReactNode;
  header?: string | React.ReactNode;
  onActionClick?: () => void;
}

export interface AccordionSingleProps {
  items: AccordionItemProps[];
  type?: RadixAccordionSingleProps["type"];
  value?: RadixAccordionSingleProps["value"];
  onChange?: RadixAccordionSingleProps["onValueChange"];
}

export interface AccordionMultipleProps {
  items: AccordionItemProps[];
  type?: RadixAccordionMultipleProps["type"];
  value?: RadixAccordionMultipleProps["value"];
  onChange?: RadixAccordionMultipleProps["onValueChange"];
}

export type AccordionProps = AccordionSingleProps | AccordionMultipleProps;
