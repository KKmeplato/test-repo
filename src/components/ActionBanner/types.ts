import type { VariantProps } from "class-variance-authority";
import { ComponentPropsWithRef, ReactNode } from "react";

import { Variant } from "lib/index";

import { actionBannerVariants } from "./utils";

export type ActionBannerVariants =
  | Extract<Variant, "default" | "critical" | "warning" | "highlight">
  | "upgrade";

export type ActionBannerVariantProps = (props?: {
  variant?: ActionBannerVariants;
}) => string;

interface BannerProps {
  variant?: ActionBannerVariants;
  children: ReactNode;
  action: {
    label: string;
    onClick?: () => void;
  };
}

export type ActionBannerProps = ComponentPropsWithRef<"div"> &
  VariantProps<typeof actionBannerVariants> &
  BannerProps;

export interface ActionBannerIconProps {
  variant?: ActionBannerVariants;
}
