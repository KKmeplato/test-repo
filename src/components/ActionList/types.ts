import { DropdownMenuProps } from "@radix-ui/react-dropdown-menu";
import { ComponentPropsWithoutRef } from "react";

import { Variant } from "lib/types";

export interface ActionItemProps
  extends Omit<ComponentPropsWithoutRef<"button">, "onClick"> {
  active?: boolean;
  variant?: Extract<Variant, "critical">;
  prefixIcon?: React.ReactNode;
  suffixIcon?: React.ReactNode;
  sideHelpText?: string | React.ReactNode;
  bottomHelpText?: string | React.ReactNode;
  onClick?: () => void;
}

export interface ActionListSection {
  title?: string;
  items: ActionItemProps[];
}

export interface ActionListProps
  extends Pick<DropdownMenuProps, "onOpenChange"> {
  items?: ActionItemProps[];
  sections?: ActionListSection[];
  activator?: React.ReactNode;
}
