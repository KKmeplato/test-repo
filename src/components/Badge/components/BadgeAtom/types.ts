import { VariantProps } from "class-variance-authority";
import { ComponentPropsWithRef } from "react";

import { Progress } from "components/Badge/types";

import { badgeAtomVariants } from "./utils";

export type BadgeAtomProps = Omit<ComponentPropsWithRef<"span">, "children"> &
  VariantProps<typeof badgeAtomVariants> & {
    progress: Progress;
  };
