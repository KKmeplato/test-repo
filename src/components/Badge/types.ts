import { VariantProps } from "class-variance-authority";
import { ComponentPropsWithRef, ReactNode } from "react";

import { Variant } from "lib/types";

import { badgeVariants } from "./utils";

export type Progress = "complete" | "partial" | "incomplete";

interface DefaultProps {
  as?: "default";
  prefixIcon?: ReactNode;
  progress?: Progress;
}

interface NotificationProps {
  as: "notification";
  progress?: Progress;
  prefixIcon?: never;
}

interface DotProps {
  as: "dot";
  progress: Progress;
  prefixIcon?: never;
}

export type BadgeVariant = Extract<
  Variant,
  "attention" | "critical" | "default" | "info" | "success" | "warning"
>;

export type BadgeVariantProps = (props?: { variant?: BadgeVariant }) => string;

export type BadgeProps = ComponentPropsWithRef<"span"> &
  VariantProps<typeof badgeVariants> &
  (DefaultProps | NotificationProps | DotProps);
