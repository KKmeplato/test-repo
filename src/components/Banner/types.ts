import type { VariantProps } from "class-variance-authority";
import { ComponentPropsWithRef, ReactNode } from "react";

import { Variant } from "lib/index";

import { bannerVariants } from "./utils";

export type BannerVariants = Extract<
  Variant,
  "default" | "success" | "critical" | "warning" | "highlight"
>;

export type BannerVariantProps = (props?: {
  variant?: BannerVariants;
  className: string | undefined;
}) => string;

export interface DefaultBannerProps {
  variant?: BannerVariants;
  hideIcon?: boolean;
  title?: string;
  children?: ReactNode;
  primaryAction?: {
    label: string;
    onClick?: () => void;
  };
  secondaryAction?: {
    label: string;
    onClick?: () => void;
  };
  onDismiss?: () => void;
  isInsideCard?: boolean;
}

export type BannerProps = ComponentPropsWithRef<"div"> &
  VariantProps<typeof bannerVariants> &
  DefaultBannerProps;

export interface BannerIconProps {
  variant?: BannerVariants;
}
