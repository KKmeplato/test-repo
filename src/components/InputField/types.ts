import {
  InputHTMLAttributes,
  ReactElement,
  TextareaHTMLAttributes
} from "react";

interface SuccessProps {
  success?: string;
  error?: never;
}

interface ErrorProps {
  success?: never;
  error?: string;
}

interface CommonInputFieldProps {
  hint?: string;
  showHintIcon?: boolean;
  label?: string;
  prefix?: string;
  prefixIcon?: ReactElement;
  preview?: boolean;
  showCharacterCount?: boolean;
  suffix?: string;
  suffixIcon?: ReactElement;
}

export interface InputProps
  extends Omit<InputHTMLAttributes<HTMLInputElement>, "onChange">,
    CommonInputFieldProps {
  multiline?: false;
  type?: "text" | "password";
  onChange?: (value: string) => void;
}

interface TextareaProps
  extends Omit<TextareaHTMLAttributes<HTMLTextAreaElement>, "onChange">,
    CommonInputFieldProps {
  multiline?: true;
  type?: "text";
  onChange?: (value: string) => void;
  min?: never;
  max?: never;
}

interface NumberInputProps
  extends Omit<InputHTMLAttributes<HTMLInputElement>, "onChange">,
    CommonInputFieldProps {
  type: "number";
  value?: number;
  prefixIcon?: never;
  suffixIcon?: never;
  multiline?: never;
  showCharacterCount?: never;
  maxLength?: never;
  onChange?: (value?: number) => void;
}

export interface StepperControlProps
  extends Pick<
    NumberInputProps,
    "max" | "min" | "value" | "disabled" | "onChange"
  > {
  min?: number;
  max?: number;
}

export type InputFieldProps = (InputProps | TextareaProps | NumberInputProps) &
  (SuccessProps | ErrorProps);
