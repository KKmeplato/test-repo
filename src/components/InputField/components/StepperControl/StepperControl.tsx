import { useStepperControl } from "src/hooks";

import { cn } from "lib/utils";

import { StepperControlProps } from "../../types";
import { StepperIcon } from "../StepperIcon";

export const StepperControl = ({
  min,
  max,
  disabled,
  value,
  onChange
}: StepperControlProps) => {
  const { increment, decrement, isIncrementDisabled, isDecrementDisabled } =
    useStepperControl({
      min,
      max,
      value,
      onChange
    });
  const className = cn(
    "h-[14px] w-[18px] cursor-pointer rounded-level-3 bg-surface-neutral text-icon select-none",
    disabled && "bg-transparent"
  );
  return (
    <div className="relative left-[6px] flex flex-col gap-0.5">
      <div
        className={cn(
          className,
          isIncrementDisabled || disabled
            ? "pointer-events-none text-icon-disabled"
            : "group-hover:bg-surface-neutral-hovered group-hover:text-icon-hovered"
        )}
        role="button"
        tabIndex={-1}
        aria-disabled={disabled || isIncrementDisabled}
        data-testid="increment"
        onClick={increment}
      >
        <StepperIcon.caretUpMinor
          className={cn(
            "rounded-level-3",
            !isIncrementDisabled && "active:bg-surface-neutral-pressed"
          )}
        />
      </div>
      <div
        className={cn(
          className,
          isDecrementDisabled || disabled
            ? "pointer-events-none text-icon-disabled"
            : "group-hover:bg-surface-neutral-hovered group-hover:text-icon-hovered"
        )}
        role="button"
        tabIndex={-1}
        aria-disabled={disabled || isDecrementDisabled}
        data-testid="decrement"
        onClick={decrement}
      >
        <StepperIcon.caretDownMinor
          className={cn(
            "rounded-level-3",
            !isDecrementDisabled && "active:bg-surface-neutral-pressed"
          )}
        />
      </div>
    </div>
  );
};
