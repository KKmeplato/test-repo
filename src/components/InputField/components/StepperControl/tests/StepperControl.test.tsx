import { render } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

import { StepperControlProps } from "../../../types";
import { StepperControl } from "../StepperControl";

const mockOnChange = vi.fn();

describe("<StepperControl />", () => {
  it.each<StepperControlProps>([
    { min: 0, max: 10, value: 5, disabled: false },
    { min: 0, max: 10, value: 5, disabled: false }
  ])("renders when min: $min, max: $max, disabled: $disabled", (props) => {
    const component = render(<StepperControl {...props} />);

    expect(component).toMatchSnapshot();
  });

  it("has increment and decrement button enabled", () => {
    const { getByTestId } = render(
      <StepperControl value={2} min={0} max={5} />
    );
    const incrementButton = getByTestId("increment");
    const decrementButton = getByTestId("decrement");

    expect(incrementButton).toHaveAttribute("aria-disabled", "false");
    expect(decrementButton).toHaveAttribute("aria-disabled", "false");
  });

  it("disables increment button when max value has reached", () => {
    const { getByTestId } = render(
      <StepperControl value={5} min={0} max={5} />
    );
    const incrementButton = getByTestId("increment");

    expect(incrementButton).toHaveAttribute("aria-disabled", "true");
  });

  it("disables decrement button when min value has reached", () => {
    const { getByTestId } = render(
      <StepperControl value={0} min={0} max={5} />
    );
    const incrementButton = getByTestId("increment");

    expect(incrementButton).toHaveAttribute("aria-disabled", "false");
  });

  it("increments the value when clicking increment spinner button", async () => {
    const { getByTestId } = render(
      <StepperControl value={0} min={0} max={5} onChange={mockOnChange} />
    );
    const incrementButton = getByTestId("increment");

    await userEvent.click(incrementButton);

    expect(mockOnChange).toHaveBeenCalledWith(1);
  });

  it("decrements the value when clicking decrement spinner button", async () => {
    const { getByTestId } = render(
      <StepperControl value={5} min={0} max={5} onChange={mockOnChange} />
    );
    const decrementButton = getByTestId("decrement");

    await userEvent.click(decrementButton);

    expect(mockOnChange).toHaveBeenCalledWith(4);
  });
});
