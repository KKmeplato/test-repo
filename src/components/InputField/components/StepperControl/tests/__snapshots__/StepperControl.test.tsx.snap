// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<StepperControl /> > renders when min: +0, max: 10, disabled: false 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="relative left-[6px] flex flex-col gap-0.5"
      >
        <div
          aria-disabled="false"
          class="h-[14px] w-[18px] cursor-pointer rounded-level-3 bg-surface-neutral text-icon select-none group-hover:bg-surface-neutral-hovered group-hover:text-icon-hovered"
          data-testid="increment"
          role="button"
          tabindex="-1"
        >
          <svg
            class="rounded-level-3 active:bg-surface-neutral-pressed"
            fill="none"
            viewBox="0 0 18 14"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M5.90169 9H12.0983C12.8493 9 13.27 8.24649 12.806 7.7324L9.70769 4.29945C9.34735 3.90018 8.65265 3.90018 8.29231 4.29945L5.19399 7.7324C4.73001 8.24649 5.15069 9 5.90169 9Z"
              fill="currentColor"
            />
          </svg>
        </div>
        <div
          aria-disabled="false"
          class="h-[14px] w-[18px] cursor-pointer rounded-level-3 bg-surface-neutral text-icon select-none group-hover:bg-surface-neutral-hovered group-hover:text-icon-hovered"
          data-testid="decrement"
          role="button"
          tabindex="-1"
        >
          <svg
            class="rounded-level-3 active:bg-surface-neutral-pressed"
            fill="none"
            viewBox="0 0 18 14"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12.0983 5L5.90169 5C5.15069 5 4.73001 5.75351 5.19399 6.2676L8.29231 9.70055C8.65265 10.0998 9.34735 10.0998 9.70769 9.70055L12.806 6.2676C13.27 5.75351 12.8493 5 12.0983 5Z"
              fill="currentColor"
            />
          </svg>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="relative left-[6px] flex flex-col gap-0.5"
    >
      <div
        aria-disabled="false"
        class="h-[14px] w-[18px] cursor-pointer rounded-level-3 bg-surface-neutral text-icon select-none group-hover:bg-surface-neutral-hovered group-hover:text-icon-hovered"
        data-testid="increment"
        role="button"
        tabindex="-1"
      >
        <svg
          class="rounded-level-3 active:bg-surface-neutral-pressed"
          fill="none"
          viewBox="0 0 18 14"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M5.90169 9H12.0983C12.8493 9 13.27 8.24649 12.806 7.7324L9.70769 4.29945C9.34735 3.90018 8.65265 3.90018 8.29231 4.29945L5.19399 7.7324C4.73001 8.24649 5.15069 9 5.90169 9Z"
            fill="currentColor"
          />
        </svg>
      </div>
      <div
        aria-disabled="false"
        class="h-[14px] w-[18px] cursor-pointer rounded-level-3 bg-surface-neutral text-icon select-none group-hover:bg-surface-neutral-hovered group-hover:text-icon-hovered"
        data-testid="decrement"
        role="button"
        tabindex="-1"
      >
        <svg
          class="rounded-level-3 active:bg-surface-neutral-pressed"
          fill="none"
          viewBox="0 0 18 14"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12.0983 5L5.90169 5C5.15069 5 4.73001 5.75351 5.19399 6.2676L8.29231 9.70055C8.65265 10.0998 9.34735 10.0998 9.70769 9.70055L12.806 6.2676C13.27 5.75351 12.8493 5 12.0983 5Z"
            fill="currentColor"
          />
        </svg>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`<StepperControl /> > renders when min: +0, max: 10, disabled: false 2`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="relative left-[6px] flex flex-col gap-0.5"
      >
        <div
          aria-disabled="false"
          class="h-[14px] w-[18px] cursor-pointer rounded-level-3 bg-surface-neutral text-icon select-none group-hover:bg-surface-neutral-hovered group-hover:text-icon-hovered"
          data-testid="increment"
          role="button"
          tabindex="-1"
        >
          <svg
            class="rounded-level-3 active:bg-surface-neutral-pressed"
            fill="none"
            viewBox="0 0 18 14"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M5.90169 9H12.0983C12.8493 9 13.27 8.24649 12.806 7.7324L9.70769 4.29945C9.34735 3.90018 8.65265 3.90018 8.29231 4.29945L5.19399 7.7324C4.73001 8.24649 5.15069 9 5.90169 9Z"
              fill="currentColor"
            />
          </svg>
        </div>
        <div
          aria-disabled="false"
          class="h-[14px] w-[18px] cursor-pointer rounded-level-3 bg-surface-neutral text-icon select-none group-hover:bg-surface-neutral-hovered group-hover:text-icon-hovered"
          data-testid="decrement"
          role="button"
          tabindex="-1"
        >
          <svg
            class="rounded-level-3 active:bg-surface-neutral-pressed"
            fill="none"
            viewBox="0 0 18 14"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12.0983 5L5.90169 5C5.15069 5 4.73001 5.75351 5.19399 6.2676L8.29231 9.70055C8.65265 10.0998 9.34735 10.0998 9.70769 9.70055L12.806 6.2676C13.27 5.75351 12.8493 5 12.0983 5Z"
              fill="currentColor"
            />
          </svg>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="relative left-[6px] flex flex-col gap-0.5"
    >
      <div
        aria-disabled="false"
        class="h-[14px] w-[18px] cursor-pointer rounded-level-3 bg-surface-neutral text-icon select-none group-hover:bg-surface-neutral-hovered group-hover:text-icon-hovered"
        data-testid="increment"
        role="button"
        tabindex="-1"
      >
        <svg
          class="rounded-level-3 active:bg-surface-neutral-pressed"
          fill="none"
          viewBox="0 0 18 14"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M5.90169 9H12.0983C12.8493 9 13.27 8.24649 12.806 7.7324L9.70769 4.29945C9.34735 3.90018 8.65265 3.90018 8.29231 4.29945L5.19399 7.7324C4.73001 8.24649 5.15069 9 5.90169 9Z"
            fill="currentColor"
          />
        </svg>
      </div>
      <div
        aria-disabled="false"
        class="h-[14px] w-[18px] cursor-pointer rounded-level-3 bg-surface-neutral text-icon select-none group-hover:bg-surface-neutral-hovered group-hover:text-icon-hovered"
        data-testid="decrement"
        role="button"
        tabindex="-1"
      >
        <svg
          class="rounded-level-3 active:bg-surface-neutral-pressed"
          fill="none"
          viewBox="0 0 18 14"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12.0983 5L5.90169 5C5.15069 5 4.73001 5.75351 5.19399 6.2676L8.29231 9.70055C8.65265 10.0998 9.34735 10.0998 9.70769 9.70055L12.806 6.2676C13.27 5.75351 12.8493 5 12.0983 5Z"
            fill="currentColor"
          />
        </svg>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
